oci/auth/aws/auth.go

```diff
@@ -34,17 +35,22 @@ import (
 	"github.com/fluxcd/pkg/oci"
 )
 
-var registryPartRe = regexp.MustCompile(`([0-9+]*).dkr.ecr.([^/.]*)\.(amazonaws\.com[.cn]*)`)
+var registryPartRe = regexp.MustCompile(`(^[a-zA-Z0-9][a-zA-Z0-9-_]*)\.dkr\.ecr(-fips)?\.([a-zA-Z0-9][a-zA-Z0-9-_]*)\.amazonaws\.com(\.cn)?$`)
```

Comment:
I took this pattern from ecr credential helper: https://github.com/awslabs/amazon-ecr-credential-helper/blob/b0fcec718dc6de8f90d6ca1cdfd9a2b35520733f/ecr-login/api/client.go#L40

I see your point, I will just add the non-capturing group as that should be enough

Reply:
I tried `([0-9+]*).dkr.ecr(?:-fips)?\.([^/.]*)\.(amazonaws\.com[.cn]*)` (added fips to the existing regex, that regex allows `".dkr.ecr.error.amazonaws.com"` and apparently the original regex also allows that as a valid string value.

Changing it to `([0-9+]+).dkr.ecr(?:-fips)?\.([^/.]*)\.(amazonaws\.com[.cn]*)` fixes it, But that might break it for some other case we are not testing already. 

I think it would be better to use the same pattern ECR is using to avoid errors